src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -103,17 +116,17 @@ public function __construct(
         $this->priceFormatter = $priceFormatter;
         $this->productColorsRetriever = $productColorsRetriever;
         $this->translator = $translator;
+        $this->hookManager = $hookManager ?? new HookManager();
```

Comment:
Isn't out of the scope of the PR :thinking:

Reply:
The only solution for adding the unit test.